pkg/api/v1/defaults.go

```diff
@@ -371,3 +371,15 @@ func SetDefaults_ScaleIOVolumeSource(obj *ScaleIOVolumeSource) {
 		obj.FSType = "xfs"
 	}
 }
+
+func SetDefaults_HugePagesVolumeSource(obj *HugePagesVolumeSource) {
+	if obj.PageSize == "" {
+		obj.PageSize = "2M"
+	}
+	if obj.MaxSize == "" {
```

Comment:
It is valid not to specify size than there is no limit being set. But I do not handle this case in volume plugin itself I have to add it in prepareMountOptions().